src/rez/shells.py

```diff
@@ -398,8 +398,10 @@ def _record_shell(ex, files, bind_rez=True, print_msg=False):
                 ex.info('')
                 ex.info('You are now in a rez-configured environment.')
                 ex.info('')
-                if system.is_production_rez_install:
-                    ex.command('rezolve context')
+                # Call python with -E directly to avoid the double cost of executing
+                # rezolve when the CLI is installed via pip.
+                rezolve = os.path.join(system.rez_bin_path, 'rezolve')
+                ex.command(f'{sys.executable} -E {rezolve} context')
```

Comment:
I'm kind of making the assumption that this is only run when using `rez-env`... Can this code ever run when solely using the API?